nat/traversal/pinger.go

```diff
@@ -171,28 +180,148 @@ func (p *Pinger) PingPeer(ip string, localPorts, remotePorts []int, initialTTL i
 		return nil, err
 	}
 
+	var pings []pingResponse
 	for res := range ch {
 		if res.err != nil {
 			log.Warn().Err(res.err).Msg("One of the pings has error")
-		} else {
+			continue
+		}
+
+		if err := ipv4.NewConn(res.conn).SetTTL(maxTTL); err != nil {
+			log.Warn().Err(res.err).Msg("Failed to set connection TTL")
+			continue
+		}
+
+		if err := sendConnACK(res.conn); err != nil {
```

Comment:
One improvement here can be to move sendConnACK into block after needed amount of pings is collected and send it concurrently waiting for response or timeout.